runner/src/main/java/io/github/rdfc/Reader.java

```diff
@@ -2,28 +2,32 @@
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
 import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 
+import java.util.logging.*;
 import com.google.protobuf.ByteString;
 
 /**
  * Channel
  */
 public class Reader implements IReader {
-    private String id;
+    private final String id;
 
     private List<StreamIter<Iter<ByteString>>> streams = new ArrayList<>();
     private List<StreamIter<ByteString>> strings = new ArrayList<>();
```

Comment:
minor: they could be final
```suggestion
private final List<StreamIter<Iter<ByteString>>> streams = new ArrayList<>();
private final List<StreamIter<ByteString>> strings = new ArrayList<>();
```